### PR TITLE
WIP: Add server unit tests

### PR DESCRIPTION
**Pull request by saschagrunert**
This PR adds unit tests to the `server` package.

### CODE REVIEW

**Comment by saschagrunert** [comment]
This big guy should be ready for review now. Sorry for the huge PR. I added some setters which are only built during test runs (`// +build test) and will be used for test data injection.

File: oci/runtime_oci.go [hunk 1]
```diff
@@ -353,7 +356,7 @@ func (r *runtimeOCI) ExecSyncContainer(c *Container, command []string, timeout i
 		args = append(args, fmt.Sprintf("%d", timeout))
 	}
 	args = append(args, "-l", logPath)
-	args = append(args, "--socket-dir-path", ContainerAttachSocketDir)
+	args = append(args, "--socket-dir-path", socketDir)
```
Comment (haircommander) [inline oci/runtime_oci.go]: do we want to do a check here to make sure socketDir != "" ?

Comment (haircommander) [reply oci/runtime_oci.go]: nevermind I didn't understand the change.

File: server/inspect_test.go [hunk 2]
```diff
@@ -1,217 +1,130 @@
...
-	}
-	_, err := s.getContainerInfo("", getContainerFunc, getInfraContainerFunc, getSandboxFunc)
-	if err == nil {
-		t.Fatal("expected an error but got nothing")
-	}
-	if err != errSandboxNotFound {
-		t.Fatalf("expected errSandboxNotFound error, got %v", err)
-	}
-}
+// The actual test suite
```
Comment (haircommander) [inline server/inspect_test.go]: nit: did you mean to leave this?

File: server/server.go [hunk 3]
```diff
@@ -287,16 +292,24 @@ func getIDMappings(config *Config) (*idtools.IDMappings, error) {
 }
 
 // New creates a new Server with options provided
-func New(ctx context.Context, config *Config) (*Server, error) {
-	if err := os.MkdirAll(oci.ContainerAttachSocketDir, 0755); err != nil {
+func New(ctx context.Context, configIface ConfigIface) (*Server, error) {
+	if configIface == nil || configIface.GetData() == nil {
+		return nil, fmt.Errorf("provided is nil")
```
Comment (haircommander) [inline server/server.go]: nit: add what is nil to make error more clear

File: server/server_inject.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// +build test
```
Comment (haircommander) [inline server/server_inject.go]: Should probably document in a comment the purpose of the *_inject.go files

File: test/testdata/cert.pem [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Comment (haircommander) [inline test/testdata/cert.pem]: What are these keys for?

**Comment by haircommander** [comment]
Did a quick once over, so far lgtm. I'll do another look later though. what's the intended plan for use of these? will they be run as CI or locally? Is there an easy way to run them locally?

**Comment by haircommander** [comment]
btw awesome work @saschagrunert !

File: server/inspect_test.go (hunk 2 above)
Comment (saschagrunert) [reply server/inspect_test.go]: Hm, I guess this comment can be removed.

File: test/testdata/cert.pem (hunk 5 above)
Comment (saschagrunert) [reply test/testdata/cert.pem]: A server test use them: :)

https://github.com/kubernetes-sigs/cri-o/blob/71adf6b4735466caf5eb6a598732d19934f83c13/server/server_test.go#L52-L65

**Comment by saschagrunert** [comment]
> Did a quick once over, so far lgtm. I'll do another look later though. what's the intended plan for use of these? will they be run as CI or locally? Is there an easy way to run them locally?

Thank you very much! In general the unit test should be able to run locally and within the CI (what implies some kind of needed flexibility). You should be able to run them via `make testunit`. :)

**Comment by saschagrunert** [comment]
Hm, it looks like we have some go version mismatch on the CI machine:

```
Detected go version: go version go1.12 linux/amd64.
Kubernetes requires go1.12.1 or greater.
Please install go1.12.1 or later.
```

**Comment by saschagrunert** [comment]
I rebased on top of the latest master branch and split up the changes into dedicated commits to make the intention more clear.

**Comment by vrothberg** [comment]
Kata seems to be flaky atm. I am having issues at another PR as well.

**Comment by saschagrunert** [comment]
Closed in favor of #2407 and all other test related PRs.

**Pull request closed without merging**